src/Components/RegisterForm.php

```diff
@@ -51,4 +51,21 @@ public function render()
             'invitation' => $this->invitationId ? Models::invitation()::findByUuid($this->invitationId) : null,
         ]);
     }
+
+    public function canSubmit(): bool
+    {
+        $requiredProperties = array_filter(
+            $this->getPublicPropertiesDefinedBySubClass(),
+            fn ($key) => in_array($key, ['code', 'name', 'username', 'email', 'password', 'password_confirmation', 'terms'], true),
+            ARRAY_FILTER_USE_KEY
+        );
```

Comment:
isn't this the same as looping over the defined array and checking for a value in `$this->{property}`?

Reply:
Not all the properties are required tho, or I misunderstand your question, so that way we only care about the ones that are required to fill in the form (e.g : username is required while subscribing to the newsletter isn't)

Reply:
isn't this the same as looping over `['code', 'name', 'username', 'email', 'password', 'password_confirmation', 'terms']` and checking for a value in `$this->{property}`?